### PR TITLE
[dagit] Treat duplicate asset definitions as a fatal error when launching materializations

### PR DESCRIPTION
**Pull request by bengotow**
### Summary & Motivation

This PR addresses https://github.com/dagster-io/dagster/issues/8748

- The existing banner warning about duplicate asset definitions is now shown in the asset sidebar (so it appears within the asset graph.) It's a bit wordy and quite large, but that might be ok. I upgraded it from "info" to "warning" to reflect how seriously we're treating this case now.

image

- Attempting to launch an asset or set of assets with a duplicate definition fails and displays an error asking you to rename your assets and pointing to the problem asset keys + repos. **This effectively prevents you from launching materializations until you rename your asset(s)!**

image

To implement this I wrote a new resolver for finding these issues given a set of asset keys. Previously doing this required fetching all asset nodes for all repos because there wasn't a great way to do the reverse lookup.

### CODE REVIEW

File: js_modules/dagit/packages/core/src/assets/AssetDefinedInMultipleReposNotice.tsx [hunk 1]
```diff
@@ -3,35 +3,39 @@ import {Alert, Box, Colors} from '@dagster-io/ui';
...
-    data?.repositoriesOrError.__typename === 'RepositoryConnection'
-      ? data.repositoriesOrError.nodes.filter(
-          (r) => r.name !== loadedFromRepo.name || r.location.name !== loadedFromRepo.location,
-        )
-      : [];
-  const otherReposWithAsset = otherRepos.filter((r) =>
-    r.assetNodes.some(
-      (a) => JSON.stringify(a.assetKey) === JSON.stringify(assetKey) && a.opNames.length,
-    ),
+  padding?: boolean;
```
[the remark this answers is not in the document]
Comment (bengotow) [reply js_modules/dagit/packages/core/src/assets/AssetDefinedInMultipleReposNotice.tsx:18]: Ahh I like it!

**Review by sryza — COMMENTED** [review]
This looks like the right experience

**Review by hellendag — APPROVED** [review]
